instill/model/v1alpha/model_service.proto

```diff
@@ -0,0 +1,202 @@
+  // LivenessResponse message.
+  // See https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+  rpc Liveness(LivenessRequest) returns (LivenessResponse) {
+    option (google.api.http) = {
+      get : "/__liveness"
+      additional_bindings : [ {get : "/health/mgmt"} ]
+    };
+  }
+
+  // Readiness method receives a ReadinessRequest message and returns a
+  // ReadinessResponse message.
+  // See https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+  rpc Readiness(ReadinessRequest) returns (ReadinessResponse) {
+    option (google.api.http) = {
+      get : "/__readiness"
+    };
+  }
+
+  // ListModelDefinition method receives a ListModelDefinitionRequest message and returns a ListModelDefinitionResponse
+  rpc ListModelDefinition(ListModelDefinitionRequest) returns (ListModelDefinitionResponse) {
+    option (google.api.http) = {
+      get : "/model-definitions"
+    };
+  }
+
+  // GetModelDefinition method receives a GetModelDefinitionRequest message and returns a GetModelDefinitionResponse
+  rpc GetModelDefinition(GetModelDefinitionRequest) returns (GetModelDefinitionResponse) {
+    option (google.api.http) = {
+      get : "/{name=model-definitions/*}"
+    };
+  }
+
+  // ListModel method receives a ListModelRequest message and returns a ListModelResponse
+  rpc ListModel(ListModelRequest) returns (ListModelResponse) {
+    option (google.api.http) = {
+      get : "/models"
+    };
+  }
+
+  // CreateModel method receives a CreateModelRequest message and returns a CreateModelResponse
+  rpc CreateModel(CreateModelRequest) returns (CreateModelResponse) {
+    option (google.api.http) = {
+      post : "/models"
+      body : "model"
+    };
+  }
+
+  // GetModel method receives a GetModelRequest message and returns a GetModelResponse
+  rpc GetModel(GetModelRequest) returns (GetModelResponse) {
+    option (google.api.http) = {
+      get : "/{name=models/*}"
+    };
+  }
+
+  // UpdateModel method receives a UpdateModelRequest message and returns a UpdateModelResponse
+  rpc UpdateModel(UpdateModelRequest) returns (UpdateModelResponse) {
+    option (google.api.http) = {
+      patch : "/{model.name=models/*}"
+      body : "model"
+    };
+  }
+
+  // DeleteModel method receives a DeleteModelRequest message and returns a DeleteModelResponse
+  rpc DeleteModel(DeleteModelRequest) returns (DeleteModelResponse) {
+    option (google.api.http) = {
+      delete : "/{name=models/*}"
+    };
+  }
+
+  // ListModelInstance method receives a ListModelInstanceRequest message and returns a ListModelInstanceResponse
+  rpc ListModelInstance(ListModelInstanceRequest) returns (ListModelInstanceResponse) {
+    option (google.api.http) = {
+      get : "/{parent=models/*}/instances"
+    };
+  }
+
+  // CreateModelInstance method receives a CreateModelInstanceRequest message and returns a CreateModelInstanceResponse
+  rpc CreateModelInstance(CreateModelInstanceRequest) returns (CreateModelInstanceResponse) {
```

Comment:
@xiaofei-du user need to use 2 API to upload a model: 1. create a model, 2. create a model instance. Or should we create a model instance when create model?

Reply:
Since we are using VCS like GitHub to manage imported instances (and we will support real-time sync from GitHub). 
I have removed create/update/delete model instance methods, only left `list/get`. It means users are not allowed to create/update/delete any model instance. They are handled by the `model-backend`.

Therefore, 
- when a user uploads a local model, `model-backend` will add the uploaded model instance into a model instance with `latest` as the model instance ID.
- when a user imports a model from GitHub, `model-backend` will fetch all tags from the repo and add all the instances to the model, with the release tag as the ID of each corresponding model instance.